lib/grpc/client/adapters/gun.ex

```diff
@@ -287,4 +321,146 @@ defmodule GRPC.Client.Adapters.Gun do
     timeout = round(timeout + jitter * timeout)
     %{retries: retries - 1, timeout: timeout}
   end
+
+  defp recv_body(conn_payload, stream_payload, opts) do
+    recv_body(conn_payload, stream_payload, "", opts)
+  end
+
+  defp recv_body(conn_payload, stream_payload, acc, opts) do
+    case recv_data_or_trailers(conn_payload, stream_payload, opts) do
+      {:data, data} ->
+        recv_body(conn_payload, stream_payload, <<acc::binary, data::binary>>, opts)
+
+      {:trailers, trailers} ->
+        {:ok, acc, GRPC.Transport.HTTP2.decode_headers(trailers)}
+
+      err ->
+        err
+    end
+  end
+
+  defp response_stream(
+         %{
+           channel: %{adapter_payload: ap},
+           response_mod: res_mod,
+           codec: codec,
+           payload: payload
+         },
+         opts
+       ) do
+    state = %{
+      adapter_payload: ap,
+      payload: payload,
+      buffer: <<>>,
+      fin: false,
+      need_more: true,
+      opts: opts,
+      response_mod: res_mod,
+      codec: codec
+    }
+
+    Stream.unfold(state, fn s -> read_stream(s) end)
+  end
+
+  defp read_stream(%{buffer: <<>>, fin: true, fin_resp: nil}), do: nil
+
+  defp read_stream(%{buffer: <<>>, fin: true, fin_resp: fin_resp} = s),
+    do: {fin_resp, Map.put(s, :fin_resp, nil)}
+
+  defp read_stream(
+         %{
+           adapter_payload: ap,
+           payload: payload,
+           buffer: buffer,
+           need_more: true,
+           opts: opts
+         } = s
+       ) do
+    case recv_data_or_trailers(ap, payload, opts) do
+      {:data, data} ->
+        buffer = buffer <> data
+        new_s = s |> Map.put(:need_more, false) |> Map.put(:buffer, buffer)
+        read_stream(new_s)
+
+      {:trailers, trailers} ->
+        trailers = GRPC.Transport.HTTP2.decode_headers(trailers)
+
+        case parse_trailers(trailers) do
+          :ok ->
+            fin_resp =
+              if opts[:return_headers] do
+                {:trailers, trailers}
+              end
+
+            new_s = s |> Map.put(:fin, true) |> Map.put(:fin_resp, fin_resp)
+            read_stream(new_s)
+
+          error ->
+            {error, %{buffer: <<>>, fin: true, fin_resp: nil}}
+        end
+
+      error = {:error, _} ->
+        {error, %{buffer: <<>>, fin: true, fin_resp: nil}}
+    end
+  end
+
+  defp read_stream(%{buffer: buffer, need_more: false, response_mod: res_mod, codec: codec} = s) do
+    case GRPC.Message.get_message(buffer) do
+      # TODO
```

Comment:
what's "TODO" here?

Reply:
IDK to be honest lol
This part was a copy and paste from `stub.ex` code. It's comes from the original author. Seems unnecessary. I'll remove this one